### PR TITLE
Advanced Site Configuration and New Hooks for Quicksilver

### LINKED ISSUE #1758: Advanced Site Configuration and New Hooks for Quicksilver

### PR DESCRIPTION
**Pull request by ari-gold**
Along with #1756 and #1408, closes #1758

## Effect
* Revise pantheon.yml and Quicksilver docs
* Add new advanced site configuration: php_version, protected_web_paths, web_root
* Two new Quicksilver hooks for MultiDev and Site Creation

### CODE REVIEW

**Comment by hengkit** [comment]
There's some broken markdown at https://advsiteco-static-docs.pantheonsite.io/docs/pantheon.yml/#nested-docroot
`For more information, please see (/docs/DOCROOT-article)`

**Comment by hengkit** [comment]
and on the `See Also` section
`Automate your Workflow with Quicksilver Platform Integration Hooks PHP Version Nested docroot`

**Comment by hengkit** [comment]
Extremely vague
`You may not be able to protect files or paths with special characters`

File: source/_docs/pantheon-yml.md [hunk 1]
```diff
@@ -1,64 +1,76 @@
...
 
-### API Version
-The current API version for `pantheon.yml` is `1`. This is the only valid value to place in this field. Additional valid values may be introduced in the future.
+## Use pantheon.yml for:
+* [Advanced Site Configuration](#advanced-site-configuration)
+  - [Nested Docroot](#nested-docroot)
+  - [PHP Version](#php-version)
+  - [Protected paths](#protected-paths)
+* [Quicksilver Platform Integration Hooks](#quicksilver-platform-integration-hooks)
+
```
Comment (rachelwhitton) [inline source/_docs/pantheon-yml.md]: Suggest removing this section - seems a little redundant to list the internal hashes here. I think the TOC provides sufficient context for using `pantheon.yml`. Defer to @nataliejeremy on final call

**Comment by ari-gold** [comment]
@rachelwhitton  I added a table, but it's not very readable. Can you please see if you can make the table more readable? Maybe start with adding the gridlines?

**Comment by rachelwhitton** [comment]
+1

**Pull request merged**